src/nanotron/data/nanoset.py

```diff
@@ -111,14 +111,26 @@ def build_nanoset_index(self) -> np.ndarray:
         dataset_index, dataset_sample_index = build_nanoset_index_helper(
             n_samples=samples_per_epoch, weights=self.dataset_weights, dataset_sizes=self.dataset_lengths
         )
-        # Shuffle the indexes the same way
-        numpy_random_state = np.random.RandomState(self.random_seed)
-        numpy_random_state.shuffle(dataset_index)
-        numpy_random_state = np.random.RandomState(self.random_seed)
-        numpy_random_state.shuffle(dataset_sample_index)
-        # Concatenate num_epochs the shuffled indexes
-        dataset_index = np.concatenate([dataset_index for _ in range(num_epochs)])
-        dataset_sample_index = np.concatenate([dataset_sample_index for _ in range(num_epochs)])
+
+        # Shuffle indices in each epoch with different random seeds and concatenate them
+        r = np.random.RandomState(self.random_seed)
+        epoch_random_seeds = r.randint(0, 2**32 - 1, num_epochs)
```

Comment:
Can we just use `self.random_seed + num_epoch` for easier reproducibility?

Reply:
Good suggestion, makes it a lot simpler and clearer what's going on. I've changed the code to incorporate this in latest commit:

https://github.com/huggingface/nanotron/blob/f060414194e02c2e22e8fa0ffd8326363917a796/src/nanotron/data/nanoset.py#L116-L123